x-pack/plugins/fleet/server/services/epm/elasticsearch/template/utils.ts

```diff
@@ -12,3 +13,33 @@ type UserSettingsTemplateName = `${TemplateBaseName}${typeof USER_SETTINGS_TEMPL
 
 export const isUserSettingsTemplate = (name: string): name is UserSettingsTemplateName =>
   name.endsWith(USER_SETTINGS_TEMPLATE_SUFFIX);
+
+// For any `constant_keyword` fields in `newMappings` that don't have a `value`, access the same field in
+// the `oldMappings` and fill in the value from there
+export const fillConstantKeywordValues = (
+  oldMappings: MappingTypeMapping,
+  newMappings: MappingTypeMapping
+) => {
+  const filledMappings = JSON.parse(JSON.stringify(newMappings)) as MappingTypeMapping;
+  const deepGet = (obj: any, keys: string[]) => keys.reduce((xs, x) => xs?.[x] ?? null, obj);
+
+  const fillEmptyConstantKeywordFields = (
+    mappings: MappingTypeMapping,
+    currentPath: string[] = []
+  ) => {
+    for (const [key, potentialField] of Object.entries(mappings)) {
```

Comment:
Is there any value to check specifically for `.properties`  instead of  going through everything?

Reply:
good idea, I updated this function